pkg/controllers/cluster/kubeconfig_controller.go

```diff
@@ -143,67 +135,37 @@ func (r *KubeconfigReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 					CertificateAuthorityData: clusterCfg.CertificateAuthorityData,
 				},
 			}}
-	}
 
-	// get oidc info from organization
-	oidc, err := r.getOIDCInfo(ctx, cluster.Namespace)
-	if err != nil {
-		kubeconfig.Status.Conditions.SetConditions(v1alpha1.TrueCondition(v1alpha1.KubeconfigReconcileFailedCondition, "OIDCFetch", err.Error()))
-		updateRequired = true
-		failed = true
-		l.Error(err, "unable to fetch oidc data")
-	} else {
-		oidcHash, err := rxhash.HashStruct(oidc)
-		if err != nil {
-			l.Error(err, "unable to hash oidc info")
-			return ctrl.Result{}, err
-		}
-		if kubeconfig.ObjectMeta.Annotations[OIDCHashAnnotation] != oidcHash {
-			updateRequired = true
-			if kubeconfig.ObjectMeta.Annotations == nil {
-				kubeconfig.ObjectMeta.Annotations = make(map[string]string)
-			}
-			kubeconfig.ObjectMeta.Annotations[OIDCHashAnnotation] = oidcHash
-
-			kubeconfig.Spec.Kubeconfig.AuthInfo = []v1alpha1.ClusterKubeconfigAuthInfoItem{
-				{
-					Name: "oidc@" + cluster.Name,
-					AuthInfo: v1alpha1.ClusterKubeconfigAuthInfo{
-						AuthProvider: clientcmdapi.AuthProviderConfig{
-							Name: "oidc",
-							Config: map[string]string{
-								"client-id":      oidc.ClientID,
-								"client-secret":  oidc.ClientSecret,
-								"idp-issuer-url": oidc.IssuerURL,
-							},
+		return nil
+	})
+
+	// collect oidc data and update kubeconfig
+	clientutil.CreateOrPatch(ctx, r.Client, &kubeconfig, func() error {
```

Comment:
Do we require the separate `clientutilCreateOrPatch(..)` from line 129. At this point all the config is ready and the patch could be performed in one operation.

Reply:
fixed by dc05d5887053145d5b8e246657f6d87309918e65